go/libraries/doltcore/sqle/tables.go

```diff
@@ -966,43 +1005,72 @@ func createIndexForTable(ctx *sql.Context, table *doltdb.Table, indexName string
 		}
 	}
 	if !doltdb.IsValidTableName(indexName) {
-		return nil, nil, nil, fmt.Errorf("invalid index name `%s` as they must match the regular expression %s", indexName, doltdb.TableNameRegexStr)
+		return nil, nil, nil, nil, fmt.Errorf("invalid index name `%s` as they must match the regular expression %s", indexName, doltdb.TableNameRegexStr)
+	}
+
+	// if an index was already created for the column set but was not generated by the user then we replace it
+	replacingIndex := false
+	existingIndex, ok := sch.Indexes().GetIndexByColumnNames(realColNames...)
+	if ok && !existingIndex.IsUserDefined() {
```

Comment:
Make sure we have tests of attempting to create a duplicate index when it was user-defined

Reply:
It's in both the go and bats tests